test/integration/dashboardTest/submitNewLocationTest.js

```diff
@@ -0,0 +1,508 @@
+// Third-party dependencies
+const chai = require('chai')
+const chaiHttp = require('chai-http')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+const { afterEach, beforeEach, describe, it } = require('mocha')
+
+// In-house dependencies
+const { helpers } = require('brave-alert-lib')
+const db = require('../../../db/db')
+const { server } = require('../../../index')
+const { clientFactory } = require('../../../testingHelpers')
+
+// Setup chai
+chai.use(chaiHttp)
+chai.use(sinonChai)
+
+const sandbox = sinon.createSandbox()
+
+const expect = chai.expect
+
+describe('dashboard.js integration tests: submitNewpLocation', () => {
+  describe('for a request that contains valid non-empty fields', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClient()
+
+      this.client = await clientFactory(db)
+
+      this.agent = chai.request.agent(server)
+
+      await this.agent.post('/login').send({
+        username: helpers.getEnvVar('WEB_USERNAME'),
+        password: helpers.getEnvVar('PASSWORD'),
+      })
+
+      this.locationid = 'unusedID'
+      this.displayName = 'locationName'
+      this.doorCoreId = 'door_coreID'
+      this.radarCoreId = 'radar_coreID'
+      this.radarType = 'XeThru'
+      this.responderPhoneNumber = '+18001231234'
+      this.twilioNumber = '+15005550006'
+      this.firmwareStateMachine = false
+      this.alertApiKey = 'myApiKey'
+      const goodRequest = {
+        locationid: this.locationid,
+        displayName: this.displayName,
+        doorCoreID: this.doorCoreId,
+        radarCoreID: this.radarCoreId,
+        radarType: this.radarType,
+        responderPhoneNumber: this.responderPhoneNumber,
+        twilioPhone: this.twilioNumber,
+        firmwareStateMachine: this.firmwareStateMachine.toString(),
+        alertApiKey: this.alertApiKey,
```

Comment:
For some reason `alertApiKey` was missing from this request. So I added it in because I think it should be there.